src/Sfa.Das.Sas.Indexer.UnitTests/ApplicationServices/Provider/Services/ProviderDataServiceTests.cs

```diff
@@ -129,6 +131,33 @@ public void FeedbackResults_FeedbackSet_Attributes_ShouldContain_DistinctList()
             Assert.AreEqual(10, provider.ProviderFeedback.Strengths.Count);
         }
 
+        [TestCase(9.9, 10)]
+        [TestCase(9.51, 10)]
+        [TestCase(9.5, 10)]
+        [TestCase(9.49, 9)]
+        [TestCase(9.2, 9)]
+        [TestCase(90.9, 91)]
+        [TestCase(90.51, 91)]
+        [TestCase(90.5, 91)]
+        [TestCase(90.49, 90)]
+        [TestCase(90.2, 90)]
+        [TestCase(0, null)]
+        [TestCase(null, null)]
+        public void ShouldRoundLearnerSatisfaction(double? finalScore, double? expected)
+        {
+            // Arrange
+            var satisfactionRateProviderResult = new SatisfactionRateProvider
+            {
+                FinalScore = finalScore
+            };
+            
+            // Act
+            var actual = _sut.GetRoundedSatisfactionRate(satisfactionRateProviderResult);
```

Comment:
Is the rounding method only public so that we can unit test it? Should it not be private, as it is only accessed from within the provider data service. The tests on the higher public method could then assert the rounding. Otherwise should the logic be moved out to somewhere separate and testable